pkg/docker/config/config_test.go

```diff
@@ -675,3 +676,170 @@ func TestAuthKeysForRef(t *testing.T) {
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b/c"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["my-registry.local"].Auth)
+			},
+		},
+	} {
+		tmpFile, err := ioutil.TempFile("", "auth.json.set")
+		require.NoError(t, err)
+		_, err = tmpFile.WriteString("{}")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpFile.Name())
+		sys := &types.SystemContext{AuthFilePath: tmpFile.Name()}
+
+		for _, registry := range tc.registries {
+			_, err := SetCredentials(sys, registry, "username", "password")
+			require.NoError(t, err)
+		}
+
+		auth, err := readJSONFile(tmpFile.Name(), false)
+		require.NoError(t, err)
+
+		tc.assert(auth)
+	}
+}
+
+func TestRemoveAuthentication(t *testing.T) {
+	testAuth := dockerAuthConfig{Auth: "ZXhhbXBsZTpvcmc="}
+	for _, tc := range []struct {
+		config     dockerConfigFile
+		registries []string
+		assert     func(dockerConfigFile)
+	}{
+		{
+			config: dockerConfigFile{
+				AuthConfigs: map[string]dockerAuthConfig{
+					"quay.io": testAuth,
+				},
+			},
+			registries: []string{"quay.io"},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 0)
+			},
+		},
+		{
+			config: dockerConfigFile{
+				AuthConfigs: map[string]dockerAuthConfig{
+					"quay.io":           testAuth,
+					"my-registry.local": testAuth,
+				},
+			},
+			registries: []string{"my-registry.local"},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 1)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io"].Auth)
+			},
+		},
+		{
+			config: dockerConfigFile{
+				AuthConfigs: map[string]dockerAuthConfig{
+					"quay.io/a/b/c":     testAuth,
+					"quay.io/a/b":       testAuth,
+					"quay.io/a":         testAuth,
+					"quay.io":           testAuth,
+					"my-registry.local": testAuth,
+				},
+			},
+			registries: []string{
+				"quay.io/a/b",
+				"quay.io",
+				"my-registry.local",
+			},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 2)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b/c"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a"].Auth)
+			},
+		},
+	} {
+
+		content, err := json.Marshal(&tc.config)
+		require.NoError(t, err)
+
+		tmpFile, err := ioutil.TempFile("", "auth.json")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpFile.Name())
+
+		_, err = tmpFile.Write(content)
+		require.NoError(t, err)
+
+		sys := &types.SystemContext{AuthFilePath: tmpFile.Name()}
+
+		for _, registry := range tc.registries {
+			err := RemoveAuthentication(sys, registry)
+			require.NoError(t, err)
+		}
+
+		auth, err := readJSONFile(tmpFile.Name(), false)
+		require.NoError(t, err)
+
+		tc.assert(auth)
+	}
+}
+
+func TestIsNamespaced(t *testing.T) {
+	for _, tc := range []struct {
+		registry string
+		expected bool
+	}{
+		{
```

Comment:
(Absolutely non-blocking: consider squeezing each entry to a single line instead of 4.)

Reply:
Fixed :+1: